src/gapstate.c

```diff
@@ -5,30 +5,88 @@
  **
  ** This file declares all variables that are considered state for the
  ** interpreter
+ ** The background is that in HPC-GAP this state has to be thread local,
+ ** that is, an instance of the state has to exist for every thread, and
+ ** global variables are unsuitable for this purpose (even if protected
+ ** from concurrent access by locks).
  **
  */
 
 #include <src/system.h>
 #include <src/gapstate.h>
 
+static GAPStateHandler GAPStateHandlers[GAPSTATE_MAX_HANDLERS];
+static Int GAPStateHandlerCount = 0;
+static Int GAPStateNext = 0; // Start of next free memory area (as offset into GAPState.StateSlots)
```

Comment:
I would not guess this meaning from the name `GAPStateNext`. Perhaps `NextFreeGAPStateSlotOffset`, or `UsedGAPStateSlotsCount`, or ...